### PR TITLE
general nice stuff for mentors

### PR DESCRIPTION
**Pull request by JohnFulpWillard**
## About The Pull Request

Gave mentors their own state so if someone somehow manages to get a mentor panel open as a non mentor, it'll close instantly (like admin panels). This isn't a major issue since mentor request manager doesn't really have anything nor is it accessible, but whatever it's better to just not, yknow?
Mentor Request panel now has a note saying who was the last person to have spoken to them, which acts like a "Was this person replied to".
Head of pseudostaff show up in mentorwho even while deadminned
Also just some minor updates to match changes that happened to admin things on TG that mentors are meant to replicate.

### CODE REVIEW

File: fulp_modules/features/mentors/mentor_state.dm [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/**
+ * tgui state: admin_state
+ *
+ * Checks that the user is an admin, end-of-story.
```
Comment (Bilbo367) [inline fulp_modules/features/mentors/mentor_state.dm:4]: shouldn't this be mentor not admin here.

Comment (JohnFulpWillard) [reply fulp_modules/features/mentors/mentor_state.dm:4]: error in copypasta

Comment (JohnFulpWillard) [inline fulp_modules/features/mentors/mentor_state.dm:4]:
```suggestion
/**
 * tgui state: mentor_state
 *
 * Checks that the user is a mentor, end-of-story.
```

File: fulp_modules/features/mentors/mentor_clientprocs.dm [hunk 2]
```diff
@@ -9,26 +8,29 @@
...
+///Anyone that isn't set to be a mentor will get nothing from this.
 /client/proc/mentor_datum_set()
 	mentor_datum = GLOB.mentor_datums[ckey]
 	if(mentor_datum)
 		mentor_datum.link_mentor_datum(ckey)
 	else if(holder)
 		new /datum/mentors(ckey)
 
-///Verifies if the client is considered a Mentor, AKA has a Mentor datum or is an Admin.
+///Retusn whether or not this client is a Mentor (Or Admin, cause they are also Mentors).
```
Comment (Momo8289) [inline fulp_modules/features/mentors/mentor_clientprocs.dm:32]:
```suggestion
/// Returns whether or not this client is a Mentor (Or Admin, cause they are also Mentors).
```

Comment (Bilbo367) [reply fulp_modules/features/mentors/mentor_clientprocs.dm:32]:
```suggestion
///Returns whether or not this client is a Mentor (Or Admin, cause they are also Mentors).
```

**Pull request merged**